backend/twitter_logger/models.py

```diff
@@ -0,0 +1,132 @@
+from functools import lru_cache, partial
+from urllib.parse import urljoin
+
+from django.conf import settings
+from django.core.urlresolvers import reverse as r
+from django.db import models
+from django.utils.translation import ugettext as _
+import tweepy
+
+from backend.base.models import LogEntry, Source
+
+
+class TwitterSource(Source):
+
+    """Connection to the Twitter API
+    """
+
+    _name = 'twitter'
+
+    @classmethod
+    def get_auth_class(cls):
+        return partial(
+            tweepy.OAuthHandler,
+            settings.TWITTER_CONSUMER_KEY,
+            settings.TWITTER_CONSUMER_SECRET,
+        )
+
+    @classmethod
+    def authorize(cls, request):
+        OAuthHandler = cls.get_auth_class()
+
+        # First step of OAuth dance: get token verifier
+        if 'oauth_verifier' not in request.GET:
+            callback_url = urljoin(
+                request.build_absolute_uri(),
+                r('sources:create_for', kwargs={'name': 'twitter'}))
+            auth = OAuthHandler(callback_url)
+            url = auth.get_authorization_url()
+            request.session['twitter_request_token'] = auth.request_token
+            return url  # Twitter authorization URL
+
+        # Second step of OAuth dance: exchange request token for access token
+        else:
+            auth = OAuthHandler()
+            auth.request_token = request.session.pop('twitter_request_token')
+            auth.get_access_token(request.GET['oauth_verifier'])
+            cls.objects.create(
+                user=request.user,
+                access_token=auth.access_token,
+                access_token_secret=auth.access_token_secret,
+            )
+            return r('sources:list')
+
+    @property
+    @lru_cache()
+    def api(self):
+        """Authorize API access with saved tokens
+        """
+        OAuthHandler = self.get_auth_class()
+        auth = OAuthHandler()
+        auth.set_access_token(self.access_token, self.access_token_secret)
+        return tweepy.API(auth)
+
+    def get_new_statuses(self):
+        """Download new user statuses from Twitter
+        """
+        try:
```

Comment:
To avoid the JS/Node style, you could do the same like this:

````python
try:
    latest = TwitterLogEntry.objects.filter(
        source=self, _type=TwitterLogEntry.TYPE_TWEET
    ).latest('datetime')
````